src/core/CCList.mli

```diff
@@ -360,15 +366,16 @@ val last : int -> 'a t -> 'a t
     [l] doesn't have that many elements). *)
 
 val head_opt : 'a t -> 'a option
-(** First element.
+(** [head_opt (x :: l)] returns [Some x] or [None] if the list is empty.
```

Comment:
`(x :: l)` may give the impression that head_opt should be call with a non-empty list only.

Reply:
Modified.